pkg/controller/migplan/registry.go

```diff
@@ -221,3 +221,116 @@ func (r ReconcileMigPlan) ensureRegistryService(client k8sclient.Client, plan *m
 
 	return nil
 }
+
+// Ensure the migration registries on both source and dest clusters have been removed
+func (r ReconcileMigPlan) ensureMigRegistriesDelete(plan *migapi.MigPlan) error {
+	var client k8sclient.Client
+
+	clusters, err := r.planClusters(plan)
+	if err != nil {
+		return err
+	}
+
+	for _, cluster := range clusters {
+		if !cluster.Status.IsReady() {
```

Comment:
Trying to understand big picture of how the concept of MigCluster readiness comes into our decision to delete registries with this PR. Can help me understand?

It looks like we're using readiness as a way to track whether we've already attempted deletion once. < Assuming this is true, I think we may get more predicatable outcomes by always checking if registry resources owned by our MigCluster currently exist anywhere, and deleting them if they do exist.

Reply:
I may be misinterpreting what the logic is intended to do, so please correct me if this is wrong.

Reply:
So this particular check was pulled from the registry and storage create code. I took the `IsReady` check to mean that we only want to attempt to execute actions on a cluster if the cluster is ready. I didn't actually look into what causes a cluster to be not ready, so that may not be the right check here. Is there some better way of saying "make sure this cluster is in a state to make calls on?"

Reply:
Related to this -- is there a better way of choosing clusters to check? The code as written won't catch registry resources allocated for this MigPlan on a cluster no longer referenced by this MigPlan -- which would happen if a user created the plan referencing ClusterA and ClusterB but later changed the ClusterB reference to ClusterC.

Reply:
I'd imagined that all of the clusters would be checked.  Any resources with the plan label would be cleaned up.

Reply:
Is there an existing method to return a list of all clusters we know about? Right now, all of the cluster-getting code we have takes a client passed in. There's a GetClient method on MigCluster, but is there anything needed to get a list of all clusters to iterate over? I think I'm getting confused by the migration vs. other clusters here -- the single GetCluster method calls `client.Get` on the passed-in client, but if I want to get all clusters, each will have its own client.

Reply:
Also, while this PR doesn't do it yet, we'll need to add a finalized that also cleans up resources on MigPlan delete. Thinking through the "remove from all clusters" bit, though, it looks like we'll also want a finalizer on MigCluster delete, since if we remove the reference to a MigCluster with plan resources allocated, there will be no way to remove these on MigPlan closing/deleting.

Reply:
> Also, while this PR doesn't do it yet, we'll need to add a finalized that also cleans up resources on MigPlan delete. Thinking through the "remove from all clusters" bit, though, it looks like we'll also want a finalizer on MigCluster delete, since if we remove the reference to a MigCluster with plan resources allocated, there will be no way to remove these on MigPlan closing/deleting.

Which resources (that won't be cleaned up on close)?

It's my understanding that finalizers that can delay/fail are frowned upon.  Perhaps someone else can comment on this?

Reply:
The concern isn't not cleaning up on close -- the concern is if a MigPlan is deleted before it's closed. We'll need a finalizer on the MigPlan that does the closing cleanup before deleting. Also, we'll need a finalizer on MigCluster that removes storage/registry resources from the cluster before removing the MigCluster CR, since once we remove the MigCluster, the MigPlan close functionality will have no way of knowing about the resources in use there.

Reply:
Agreed with all the concerns and perhaps the finalizer is just a necessary evil.